gui/app.py

```diff
@@ -23,8 +24,9 @@ def __init__(self):
         self.recruitment_frame = RecruitmentFrame(self)
         self.construction_frame = ConstructionFrame(self)
         self.settings_frame = SettingsFrame(self)
+        self.api_key_frame = APIKeyFrame(self)
 
-        self.current_frame = self.lookup_frame
+        self.current_frame = self.api_key_frame
```

Comment:
Add a check/function here that verifies if the API key is already saved & if it's valid. If any of those conditions aren't met, then show the api key frame.